packages/telemetry/lib/pluggable-instrumentations.js

```diff
@@ -0,0 +1,108 @@
+const { MongoDBInstrumentation } = require('@opentelemetry/instrumentation-mongodb')
+const { MongooseInstrumentation } = require('@opentelemetry/instrumentation-mongoose')
+const { MySQL2Instrumentation } = require('@opentelemetry/instrumentation-mysql2')
+const { MySQLInstrumentation } = require('@opentelemetry/instrumentation-mysql')
+const { NestInstrumentation } = require('@opentelemetry/instrumentation-nestjs-core')
+const { NetInstrumentation } = require('@opentelemetry/instrumentation-net')
+const { PgInstrumentation } = require('@opentelemetry/instrumentation-pg')
+const { PinoInstrumentation } = require('@opentelemetry/instrumentation-pino')
+const { RedisInstrumentation: RedisInstrumentationV2 } = require('@opentelemetry/instrumentation-redis')
+const { RedisInstrumentation: RedisInstrumentationV4 } = require('@opentelemetry/instrumentation-redis-4')
+const { RestifyInstrumentation } = require('@opentelemetry/instrumentation-restify')
+const { RouterInstrumentation } = require('@opentelemetry/instrumentation-router')
+const { SocketIoInstrumentation } = require('@opentelemetry/instrumentation-socket.io')
+const { TediousInstrumentation } = require('@opentelemetry/instrumentation-tedious')
+const { UndiciInstrumentation } = require('@opentelemetry/instrumentation-undici')
+const { WinstonInstrumentation } = require('@opentelemetry/instrumentation-winston')
+
+// This list is from: 'https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/metapackages/auto-instrumentations-node/src/utils.ts#L96
+const InstrumentationMap = {
+  amqplib: AmqplibInstrumentation,
+  'aws-lambda': AwsLambdaInstrumentation,
+  'aws-sdk': AwsInstrumentation,
+  bunyan: BunyanInstrumentation,
+  'cassandra-driver': CassandraDriverInstrumentation,
+  connect: ConnectInstrumentation,
+  cucumber: CucumberInstrumentation,
+  dataloader: DataloaderInstrumentation,
+  dns: DnsInstrumentation,
+  express: ExpressInstrumentation,
+  fastify: FastifyInstrumentation,
+  fs: FsInstrumentation,
+  'generic-pool': GenericPoolInstrumentation,
+  graphql: GraphQLInstrumentation,
+  grpc: GrpcInstrumentation,
+  hapi: HapiInstrumentation,
+  http: HttpInstrumentation,
+  ioredis: IORedisInstrumentation,
+  kafkajs: KafkaJsInstrumentation,
+  knex: KnexInstrumentation,
+  koa: KoaInstrumentation,
+  'lru-memoizer': LruMemoizerInstrumentation,
+  memcached: MemcachedInstrumentation,
+  mongodb: MongoDBInstrumentation,
+  mongoose: MongooseInstrumentation,
+  mysql2: MySQL2Instrumentation,
+  mysql: MySQLInstrumentation,
+  'nestjs-core': NestInstrumentation,
+  net: NetInstrumentation,
+  pg: PgInstrumentation,
+  pino: PinoInstrumentation,
+  redis: RedisInstrumentationV2,
+  'redis-4': RedisInstrumentationV4,
+  restify: RestifyInstrumentation,
+  router: RouterInstrumentation,
+  'socket.io': SocketIoInstrumentation,
+  tedious: TediousInstrumentation,
+  undici: UndiciInstrumentation,
+  winston: WinstonInstrumentation,
+}
```

Comment:
Should we be lazy-loading these? That's a lot of files to load which likely most would go unused.